subsys/net/lib/mqtt/mqtt_pkt.c

```diff
@@ -815,7 +815,7 @@ int mqtt_unpack_suback(uint8_t *buf, uint16_t length, uint16_t *pkt_id,
 	*pkt_id = ntohs(val_u16);
 	offset += PACKET_ID_SIZE;
 
-	*items = length - offset;
+	*items = rlen - PACKET_ID_SIZE;
```

Comment:
This line looks good, but wanted to make a note about the commit message: what I remember seeing was a SUBACK and a PINGRESP combined in a single net_buf (IIRC they were in the same IPv6 packet). So first SUBACK (SUBACK variable header + one QoS item), the next MQTT packet had length 2 (0xd000 == PINGRESP), leading to 3 items (for the one QoS item and the two bytes in the PINGRESP). Did you see the two-SUBACKS-in-a-single-packet in your own testing?